### PR TITLE
Valgrind test fixes

### PR DESCRIPTION
**Pull request by richardcohen**
Some Valgrind fixes that are completely isolated to test code

### CODE REVIEW

File: libgnucash/engine/test-core/test-engine-stuff.cpp [hunk 1]
```diff
@@ -1504,10 +1499,8 @@ get_random_guids(int max)
...
 free_random_guids(GList *guids)
 {
-    GList *node;
-
-    for (node = guids; node; node = node->next)
-        g_free (node->data);
+    for (GList *node = guids; node; node = node->next)
+        guid_free (static_cast<GncGUID*>(node->data));
 
     g_list_free (guids);
```
Comment (christopherlam) [inline libgnucash/engine/test-core/test-engine-stuff.cpp]:
```suggestion
    g_list_free_full (guids, (GDestroyNotify)guid_free);
```

File: libgnucash/engine/test-core/test-engine-stuff.cpp [hunk 2]
```diff
@@ -2032,7 +2022,7 @@ make_trans_query (Transaction *trans, TestQueryTypes query_types)
         xaccQueryAddAccountGUIDMatch (q, list, QOF_GUID_MATCH_ANY, QOF_QUERY_AND);
 
         for (node = list; node; node = node->next)
-            g_free (node->data);
+            guid_free (static_cast<GncGUID*>(node->data));
         g_list_free (list);
```
Comment (christopherlam) [inline libgnucash/engine/test-core/test-engine-stuff.cpp]: oops sorry this one too!

Comment (richardcohen) [reply libgnucash/engine/test-core/test-engine-stuff.cpp]: My fault -- I should have checked more carefully. Thanks.

**Comment by jralls** [comment]
Thanks!

**Pull request merged**